crates/db_schema/src/source/moderator.rs

```diff
@@ -181,6 +182,26 @@ pub struct ModAddCommunityForm {
   pub removed: Option<bool>,
 }
 
+#[derive(Clone, Queryable, Identifiable, PartialEq, Debug, Serialize)]
+#[table_name = "mod_transfer_community"]
+pub struct ModTransferCommunity {
+  pub id: i32,
+  pub mod_person_id: PersonId,
+  pub other_person_id: PersonId,
+  pub community_id: CommunityId,
+  pub removed: Option<bool>,
+  pub when_: chrono::NaiveDateTime,
+}
```

Comment:
There are a lot of duplicate fields in all of these structs, and probably even more to handle them all. I think you could simplify it by implementing it in the following way:
```
/// common fields for all mod actions
struct ModAction {
    id: i32,
    mod_persion_id: PersonId,
    other_person_id: PersonId,
    community_id: CommunityId,
    mod_action_data: ModActionData,
}

struct ModTransferCommunity {
    removed: Option<bool>,
}

enum ModActionData {
    ModTransferCommunity(ModTransferCommunity),
    ModAddCommunity(ModAddCommunity),
    ...
}
```

Reply:
Not only do I really dislike using enums to replace explicit data types like this, but diesel needs those macros on each specific table.

Reply:
Alright, but I dislike duplicate code and the db code probably has a lot of it.